EssentialsChat/src/main/java/com/earth2me/essentials/chat/processing/SignedChatHandler.java

```diff
@@ -0,0 +1,120 @@
+package com.earth2me.essentials.chat.processing;
+
+import com.earth2me.essentials.Essentials;
+import com.earth2me.essentials.I18n;
+import com.earth2me.essentials.chat.EssentialsChat;
+import com.earth2me.essentials.utils.VersionUtil;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.EventPriority;
+import org.bukkit.event.Listener;
+import org.bukkit.event.player.AsyncPlayerChatEvent;
+import org.bukkit.event.player.AsyncPlayerChatPreviewEvent;
+import org.bukkit.plugin.PluginManager;
+
+public class SignedChatHandler extends AbstractChatHandler {
+
+    public SignedChatHandler(Essentials ess, EssentialsChat essChat) {
+        super(ess, essChat);
+    }
+
+    public boolean tryRegisterListeners() {
+        if (VersionUtil.getServerBukkitVersion().isLowerThan(VersionUtil.v1_19_1_R01)) {
+            return false;
+        }
+
+        try {
+            final Class<?> previewClass = Class.forName("org.bukkit.event.player.AsyncPlayerChatPreviewEvent");
+            if (!AsyncPlayerChatEvent.class.isAssignableFrom(previewClass)) {
+                essChat.getLogger().severe(I18n.tl("essChatNoSecureMsg", essChat.getDescription().getVersion()));
+                return false;
+            }
```

Comment:
Is this check needed? Why wouldn't this be the case?

Reply:
This is a draft API, and it's entirely possible Spigot might separate these events in the future (ie to provide the player's signature in the chat event, which wouldn't be relevant in the preview event). This just handles that possibility without blowing up.